File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/linux/LinuxButton.java
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.linux;

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.Button;

public class LinuxButton implements Button {
    @Override
    public void onClick() {
        System.out.println("linux button click");
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/linux/LinuxButton.kt
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.linux

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.Button

class LinuxButton: Button {
    override fun onClick() {
        println("windows button clicked")
    }
}